test/base/UpgradeableProxyUpgradeToTest.js

```diff
@@ -0,0 +1,128 @@
+// JS Libraries
+const withData = require("leche").withData;
+const { t } = require("../utils/consts");
+
+// Smart contracts
+const AdminUpgradeabilityProxy = artifacts.require("./base/UpgradeableProxy.sol");
+const UpgradableV1 = artifacts.require("./mock/upgradable/UpgradableV1.sol");
+const UpgradableV2 = artifacts.require("./mock/upgradable/UpgradableV2.sol");
+
+const DAI = artifacts.require("./mock/token/DAIMock.sol");
+
+contract("UpgradeableProxyUpgradeToTest", function(accounts) {
+    let dai;
+    let v1LibraryInstance;
+    let v2LibraryInstance;
+    let v1InitData;
+    let initValue = 43;
+
+    before(async () => {
+        dai = await DAI.new()
+    })
+
+    beforeEach("Setup for each test", async () => {
+        v1LibraryInstance = await UpgradableV1.new();
+        v2LibraryInstance = await UpgradableV2.new();
+
+        v1InitData = v1LibraryInstance.contract.methods.initialize(initValue).encodeABI();
```

Comment:
I see we need to encode the ABI and we have several contracts using ABI encoder v2, do we know if there will be no problem using V2 enconder?

Reply:
It is only necessary to encode the ABI of the `initialize`er functions on upgradeable contracts that we want to deploy and then immediately call the initialize function on them. Otherwise if you don't want to deploy the contract and call a function on it at the same time, you can just pass 0 bytes (`0x`) as the 3rd argument to the upgradeable proxy as you can see in this test a few lines down on line 38-39:
```
                const initData = initAfter ? "0x" : v1InitData;
                const proxy = await AdminUpgradeabilityProxy.new(v1LibraryInstance.address, admin, initData);
```

Reply:
Yes, my question is about making sure we don't end up having encoding issues on the contracts that use ABIEncoderV2. For example InterestConsensus uses ABIEncoderV2 and we will make it upgradeable. In the case we want to call initialize() during deployment (which makes sense) we will call initialize(params).encodeABI(). Is there something we can do to validate we won't have encoding issues on ABIEncoderV2 contracts?